universe/aws/client/factory.go

```diff
@@ -18,23 +17,12 @@ import (
 )
 
 type ClientFactory struct {
-	SharedCredentialsPath string
-
 	openTelemetry tracing.DeferredTracerProvider
 }
 
 func (cf ClientFactory) NewWithCreds(ctx context.Context, optFns ...func(*config.LoadOptions) error) (aws.Config, error) {
-	if tokenFile := os.Getenv("AWS_WEB_IDENTITY_TOKEN_FILE"); tokenFile == "" {
-		if cf.SharedCredentialsPath == "" {
-			return aws.Config{}, errors.New("when running without universe/aws/irsa, aws credentials are required to be set")
-		}
-
-		core.Log.Printf("[aws/client] using shared credentials at %q", cf.SharedCredentialsPath)
-
-		optFns = append(optFns, config.WithSharedCredentialsFiles([]string{cf.SharedCredentialsPath}))
-	} else {
-		core.Log.Printf("[aws/client] using web identity credentials at %q", tokenFile)
-	}
+	tokenFile := os.Getenv("AWS_WEB_IDENTITY_TOKEN_FILE")
```

Comment:
The error path from above should be maintained, i.e. if the env var is not set, then we should fail.

Reply:
Done.